### PR TITLE
Wi-Fi statistics Reset command

### PR DESCRIPTION
**Pull request by kapbh**
Add reset command and net management api to reset Wi-Fi statistics.

### CODE REVIEW

File: subsys/net/l2/wifi/wifi_mgmt.c [hunk 1]
```diff
@@ -515,6 +515,20 @@ static int wifi_iface_stats(uint32_t mgmt_request, struct net_if *iface,
 	return wifi_mgmt_api->get_stats(dev, stats);
 }
 NET_MGMT_REGISTER_REQUEST_HANDLER(NET_REQUEST_STATS_GET_WIFI, wifi_iface_stats);
+
+static int wifi_iface_stats_reset(uint32_t mgmt_request, struct net_if *iface,
+		void *data, size_t len)
```
Comment (rlubos) [inline subsys/net/l2/wifi/wifi_mgmt.c:520]: Alignment seems off here.

File: subsys/net/l2/wifi/wifi_shell.c [hunk 2]
```diff
@@ -887,10 +887,22 @@ static int cmd_wifi_stats(const struct shell *sh, size_t argc, char *argv[])
 	struct net_stats_wifi stats = { 0 };
 	int ret;
 
-	ret = net_mgmt(NET_REQUEST_STATS_GET_WIFI, iface,
-				&stats, sizeof(stats));
-	if (!ret) {
-		print_wifi_stats(iface, &stats, sh);
+	if (argc == 1) {
+		ret = net_mgmt(NET_REQUEST_STATS_GET_WIFI, iface,
+			&stats, sizeof(stats));
```
Comment (rlubos) [inline subsys/net/l2/wifi/wifi_shell.c:892]: Ditto

**Review by krish2718 — DISMISSED** [review]

File: subsys/net/l2/wifi/wifi_shell.c [hunk 3]
```diff
@@ -2042,7 +2054,7 @@ SHELL_STATIC_SUBCMD_SET_CREATE(wifi_commands,
 		    "[-h, --help] : Print out the help for the scan command.\n",
 		  cmd_wifi_scan,
 		  1, 8),
-	SHELL_CMD_ARG(statistics, NULL, "Wi-Fi interface statistics.\n", cmd_wifi_stats, 1, 0),
+	SHELL_CMD_ARG(statistics, NULL, "Wi-Fi interface statistics.\n", cmd_wifi_stats, 1, 1),
```
Comment (krish2718) [inline subsys/net/l2/wifi/wifi_shell.c:2057]: will the help show the possible options? or may be migrate to getopt?

Comment (kapbh) [reply subsys/net/l2/wifi/wifi_shell.c:2057]: Will add getopt for showing options.

File: subsys/net/l2/wifi/wifi_shell.c (hunk 2 above)
Comment (kapbh) [reply subsys/net/l2/wifi/wifi_shell.c:892]: It was put as it is to get Wi-Fi statistics.

**Review by krish2718 — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**